util/add-depends.pl

```diff
@@ -71,14 +71,33 @@
     map { catfile($abs_blddir, $_) => $_ }
     keys %{$unified_info{generate}};
 
+# GCC 4.1 prints the object twice in the dependency file. e.g.:
+
+# foo.o foo.o: foo.c
```

Comment:
May I know the reason for why it prints twice?

Is there any way we can avoid printing it twice? So that we don't need this change?

Reply:
That's probably an "oops" in that gcc version. It's not *wrong* (I assume `make` would just ignore the duplicate).

It's nothing we can change, really... or avoid, other than advising people to avoid that version

Reply:
I can reproduce this (with gcc 4.1) with:
```
$ gcc  -MMD -MF /dev/stdout -MT foo.o -o foo.o -c foo.c
foo.o foo.o: foo.c foo.h
```

If I remove `-MT` or `-o` then the dup entry is gone.

Reply:
Considering that we throw away the target anyway in `add-depend.pl` (we have to regenerate it anyway to restore proper directory information), it should be pretty safe to drop `-MT {arg}`.

That might actually be the better idea...  are you willing to experiment with that?

Reply:
Incidentally, dropping `-MT` will also fix a problem on NonStop (see NOTES-NONSTOP.md)

Reply:
Dropping `-MT` fixes the issue for me, and the d file looks ok (same as with `-MT` except the duplicate object).

I'll update this PR.